outputs/otlp.go

```diff
@@ -0,0 +1,150 @@
+
+	sc := trace.SpanContext{}.WithTraceID(traceID)
+	ctx := trace.ContextWithSpanContext(context.Background(), sc)
+
+	tracer := getTracerProvider().Tracer("falco-event")
+	_, span := tracer.Start(
+		ctx,
+		falcopayload.Rule,
+		trace.WithTimestamp(startTime),
+		trace.WithSpanKind(trace.SpanKindServer))
+
+	span.SetAttributes(attribute.String("uuid", falcopayload.UUID))
+	span.SetAttributes(attribute.String("source", falcopayload.Source))
+	span.SetAttributes(attribute.String("priority", falcopayload.Priority.String()))
+	span.SetAttributes(attribute.String("rule", falcopayload.Rule))
+	span.SetAttributes(attribute.String("output", falcopayload.Output))
+	span.SetAttributes(attribute.String("hostname", falcopayload.Hostname))
+	span.SetAttributes(attribute.StringSlice("tags", falcopayload.Tags))
+	for k, v := range falcopayload.OutputFields {
+		span.SetAttributes(attribute.String(k, fmt.Sprintf("%v", v)))
+	}
+	// span.AddEvent("falco-event")
+	span.End(trace.WithTimestamp(endTime))
+
+	if c.Config.Debug {
+		log.Printf("[DEBUG] : OTLP Traces - payload generated successfully for traceid=%s", span.SpanContext().TraceID())
+	}
+
+	return &span
+}
+
+// OTLPPost generates an OTLP trace _implicitly_ via newTrace() by
+// calling OTEL SDK's tracer.Start() --> span.End(), i.e. no need to explicitly
+// do a HTTP POST
+func (c *Client) OTLPTracesPost(falcopayload types.FalcoPayload) {
+	c.Stats.OTLPTraces.Add(Total, 1)
+
+	trace := c.newTrace(falcopayload)
+	if trace == nil {
+		go c.CountMetric(Outputs, 1, []string{"output:otlptraces", "status:error"})
+		c.Stats.OTLPTraces.Add(Error, 1)
+		c.PromStats.Outputs.With(map[string]string{"destination": "otlptraces", "status": Error}).Inc()
+		log.Printf("[ERROR] : OLTP Traces - Error generating trace")
+		return
+	}
+	// Setting the success status
+	go c.CountMetric(Outputs, 1, []string{"output:otlptraces", "status:ok"})
+	c.Stats.OTLPTraces.Add(OK, 1)
+	c.PromStats.Outputs.With(map[string]string{"destination": "otlptraces", "status": OK}).Inc()
+}
+
+const (
+	templateOption       = "missingkey=zero"
+	containerTemplateStr = `{{.container_id}}`
+)
+
+var (
+	containerTemplate = template.Must(template.New("").Option(templateOption).Parse(containerTemplateStr))
+)
+
+func sanitizeOutputFields(falcopayload types.FalcoPayload) map[string]interface{} {
+	ret := make(map[string]interface{})
+	for k, v := range falcopayload.OutputFields {
+		k := strings.ReplaceAll(k, ".", "_")
+		ret[k] = v
+	}
+	return ret
+}
+
+func renderTraceIDFromTemplate(falcopayload types.FalcoPayload, config *types.Configuration) (string, string) {
+	tplStr := config.OTLP.Traces.TraceIDHash
+	tpl := config.OTLP.Traces.TraceIDHashTemplate
+	outputFields := sanitizeOutputFields(falcopayload)
+	// Default to container.id `{{.container_id}}` as templating "seed" to generate traceID.
+	if tplStr == "" {
+		tpl, tplStr = containerTemplate, containerTemplateStr
+	}
```

Comment:
Once again, we can't rely on that field as some rules don't output it or it's not available for plugins.

Can we allow the users to specify multiple seeds? And use them in an order.

k8s.pod.name + container.id > k8s.pod.name > container.id > container.name > ???

Still remains the case with events without these fields. We can also choose to explicitly consider only the syscalls events.

Reply:
Understood and acknowledged already, my take is that by exposing it as a env var, we'd let users to adapt it to their own setup (including plugins).
Still, +1 on changing the default to something that would have a "wider coverage", IMO we could put all the above into the single existing default, as we're here chasing a seed that we'll hash into the `traceID`, to set as _the_ rooting value (for the spans).

What do you think about just changing the default? ->
* from: `{{default_id}}`
* to: `{{k8s_ns_name}}{{k8s_pod_name}}{{container_name}}{{container_id}}`

NOTE that we can't _easily_ use dots in `gotemplate`, as it'd assume fields hierarchy and make the template _really_ cumbersome (need to use gotemplate nested `index` calls, etc).

Alternatively, I could implement your suggestion as a multi-valued string e.g. separated by `;` (single envvar / config field), but IMO is not needed as per the "aggregated" nature of the template rendering.

Reply:
Actually went ahead an implemented the above "expanded" default template string in fe6b950c, for your consideration.